src/main.js

```diff
@@ -2,18 +2,107 @@
+var descriptor1 = document.querySelector('.tagline-1');
+var descriptor2 = document.querySelector('.tagline-2');
 // Add your event listeners here 👇
+document.querySelector('.random-cover-button').addEventListener('click', function(){
+  var randomCover = createRandomCover();
+  mainImage.src = randomCover.cover;
+  title.innerHTML = randomCover.title;
+  // console.log(title.innerText)
+  descriptor1.innerHTML = randomCover.tagline1;
+  descriptor2.innerHTML = randomCover.tagline2;
+});
 
 
 // Create your event handlers and other functions here 👇
 
+function getRandomIndex(covers) {
+  return Math.floor(Math.random() * covers.length);
+}
 
-// We've provided one function to get you started
-function getRandomIndex(array) {
-  return Math.floor(Math.random() * array.length);
+function createRandomCover(){
+  var randomImage = covers[getRandomIndex(covers)];
+  var randomTitle = titles[getRandomIndex(titles)];
+  var randomDescriptor1 = descriptors[getRandomIndex(descriptors)];
+  var randomDescriptor2 = descriptors[getRandomIndex(descriptors)];
+  var randomCover = new Cover(randomImage, randomTitle, randomDescriptor1, randomDescriptor2)
+  return randomCover
 }
+
+
+//Change visibility of the form
+document.querySelector('.make-new-button').addEventListener('click',function(){
+  document.getElementsByClassName('random-cover-button')[0].style.display = 'none';
+  document.getElementsByClassName('save-cover-button')[0].style.display = 'none';
+  document.getElementsByClassName('home-button')[0].style.display = 'block';
+  document.getElementsByClassName('view home-view')[0].style.display = 'none';
+  document.getElementsByClassName('view form-view hidden')[0].style.display = 'block';
+})
+
+//Saved covers button, iteration 2 
+document.querySelector('.view-saved-button').addEventListener('click',function(){
+  document.getElementsByClassName('random-cover-button')[0].style.display = 'none';
+  document.getElementsByClassName('save-cover-button')[0].style.display = 'none';
+  document.getElementsByClassName('home-button')[0].style.display = 'block';
+  document.getElementsByClassName('view home-view')[0].style.display = 'none';
+  document.getElementsByClassName('view form-view hidden')[0].style.display = 'none';
+  document.getElementsByClassName('saved-covers-section')[0].style.display = 'block';
+})
+//Home button, iteration 2 
+document.querySelector('.home-button').addEventListener('click',function(){
+  document.getElementsByClassName('random-cover-button')[0].style.display = 'block';
+  document.getElementsByClassName('save-cover-button')[0].style.display = 'block';
+  document.getElementsByClassName('home-button')[0].style.display = 'none';
+  document.getElementsByClassName('view home-view')[0].style.display = 'block';
+  document.getElementsByClassName('view form-view hidden')[0].style.display = 'none';
+  document.getElementsByClassName('saved-covers-section')[0].style.display = 'none';
+})
```

Comment:
This is all really straightforward and clean! I think we could make some of the reusable and DRY things up a bit. Let's go over this on Sunday 🙂